src/persist-client/src/read.rs

```diff
@@ -624,31 +673,26 @@ where
         // Drop the encoded representation as soon as we can to reclaim memory.
         drop(value);
 
-        let mut ret = Vec::new();
+        // WIP this is really subtle and possibly wrong, justify!
```

Comment:
@aljoscha I was talking to paul yesterday about this. we currently have two "kinds" of batches in persist. one is a batch written directly by a client, which might be registered with truncated bounds and will always have a since of the minimum timestamp. the other is a batch written by compaction, which always is registered with bounds that exactly match the data in blob and the since might be downgraded. we need to perform this filtering logic on the first but not the second (and in the general case, couldn't perform it because the since might have erased the distinction in the original timestamps).

i'm not sure what to do about this/how to structure this code in a satisfying way, but it feels a bit funky right now, so calling it out and I'd love your thoughts

Reply:
Not familiar enough to know if the current `since.less_equal(lower)` behavior is expected to be a guarantee going forward, but agreed that it feels subtle... feels like we want one more bit of information

Reply:
wrote a much bigger comment here and restructured the code a bit. i still don't love it, so happy to accept any suggestions if you have them!

Reply:
I think with the latest change (big comment) this is fine to merge/I didn't come up with anything better after pondering it for a while. Though I agree that it's funky!